### PR TITLE
[GH-6184] - Make able to disable tabs inside the MoreViewController [iOS]

### LINKED ISSUE #6184: [Bug] Invalid TabBar.Items access in iOS ShellItemRenderer
### Description

iOS ShellItemRenderer accesses [TabBar.Items](https://github.com/xamarin/Xamarin.Forms/blob/master/Xamarin.Forms.Platform.iOS/Renderers/ShellItemRenderer.cs#L259) to enable / disable items. If there are more shell sections than maxTabs, it could lead to an index out of range exception because they are aggregated in a single "show more" item.

### Steps to Reproduce

1. Create shell project
2. Create additional shell sections
3. Set shell section to IsEnabled false to trigger array access
4. Start app

```xml
<!-- Your Pages -->
<ShellItem>
    <ShellSection Title="Browse" Icon="tab_feed.png">
        <ShellContent ContentTemplate="{DataTemplate local:ItemsPage}" />
    </ShellSection>
    <ShellSection Title="About" Icon="tab_about.png">
        <ShellContent ContentTemplate="{DataTemplate local:AboutPage}" />
    </ShellSection>
    <ShellSection Title="Browse" Icon="tab_feed.png">
        <ShellContent ContentTemplate="{DataTemplate local:ItemsPage}" />
    </ShellSection>
    <ShellSection Title="About" Icon="tab_about.png">
        <ShellContent ContentTemplate="{DataTemplate local:AboutPage}" />
    </ShellSection>
    <ShellSection Title="Browse" Icon="tab_feed.png">
        <ShellContent ContentTemplate="{DataTemplate local:ItemsPage}" />
    </ShellSection>
    <ShellSection Title="About" Icon="tab_about.png">
        <ShellContent ContentTemplate="{DataTemplate local:AboutPage}" />
    </ShellSection>
    <ShellSection Title="Browse" Icon="tab_feed.png">
        <ShellContent ContentTemplate="{DataTemplate local:ItemsPage}" />
    </ShellSection>
    <ShellSection Title="About" Icon="tab_about.png" IsEnabled="False">
        <ShellContent ContentTemplate="{DataTemplate local:AboutPage}" />
    </ShellSection>
</ShellItem>
```

### Expected Behavior

No exception. You might also have to fiddle with the `UIMoreNavigationController` to disable particular items there.

### Actual Behavior

Exception.

### PR DESCRIPTION
**Pull request by pictos**
### Description of Change ###

Now the XF doesn't throw an exception when a TabBarItem in MoreViewController has IsEnabled = false.

### Issues Resolved ### 

- partially fixes #6184 

### API Changes ###

Added:
 - local function `void SetEnabledForMoreController()` in `ShellItemRenderer.CreateTabRenderers()`

### Platforms Affected ### 

- iOS

### Behavioral/Visual Changes ###

None

### Before/After Screenshots ### 
Before:
throws Exception

Now:
image

### Testing Procedure ###

Create a Shell Bottom Tab application with more than 5 tabs, and set `IsEnabled` property to false.

### PR Checklist ###

- [ ] Targets the correct branch
- [ ] Tests are passing (or failures are unrelated)

### CODE REVIEW

**Comment by PureWeen** [comment]
@pictos can you rebase this against master?

**Comment by pictos** [comment]
@PureWeen, I did the rebase.

**Comment by PureWeen** [comment]
@pictos it doesn't look like it was done quite correctly

if you look at the changed files
https://github.com/xamarin/Xamarin.Forms/pull/9781/files

there is a bunch of stuff there

**Comment by pictos** [comment]
I saw I'm looking on that 😅

**Comment by pictos** [comment]
@PureWeen, can you look to see if I fixed it?

**Comment by PureWeen** [comment]
@pictos 👍 

Kicked off the UI Tests

**Comment by pictos** [comment]
@PureWeen this error is related to my changes?

**Comment by PureWeen** [comment]
@pictos looking at your branch it doesn't look like it's been rebased to master. Your master branch looks updated though

when you're on your gh-6184 try 

git rebase master

And then do a force push

**Comment by pictos** [comment]
@PureWeen, one more try, and fingers crossed 🤞🏾

**Comment by PureWeen** [comment]
@pictos it looks like `GitHubIssue6184_apple_iphone_11_pro_13_3_1`

Is failing on all our iOS UI tests

**Comment by pictos** [comment]
@PureWeen, the error from CI is: 
`System.Exception : Error while performing Tap(Marked("Issue 5"))`
`----> System.Exception : Unable to tap element. Query for Marked("Issue 5") gave no results.`

can I ignore the UITest? I would not able to run it on my machine, but the test is passing manually.

**Comment by pictos** [comment]
@PureWeen, any update on this?

**Comment by pictos** [comment]
@PureWeen I figured out way this test is failing. It's because the automated test can't click on More tab, do you know how to set an AutomationId for it?

**Comment by pictos** [comment]
@PureWeen, I fixed the UITest for it, can you run it, please?

**Comment by pictos** [comment]
@PureWeen, the tests that failed don't seem to be related to this PR, is that correct?

File: Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Issue6184.xaml.cs [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+		}
+
+		protected override void Init()
+		{
+		}
+
+#if UITEST
+#if !(__IOS__)
+		[Ignore("This issue is just for iOS")]
+#endif
```
Comment (PureWeen) [inline Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Issue6184.xaml.cs:36]:
```suggestion
#if UITEST && __IOS__
```

File: Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Issue6184.xaml [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+             xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
+             xmlns:d="http://xamarin.com/schemas/2014/forms/design"
+             xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
+    xmlns:local="clr-namespace:Xamarin.Forms.Controls.Issues"
+    mc:Ignorable="d"
+             x:Class="Xamarin.Forms.Controls.Issues.Issue6184">
+    <ShellItem>
+        <ShellSection Title="Issue 1" IsEnabled="False">
+            <ShellContent>
+                <local:Issue1900/>
```
Comment (PureWeen) [inline Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Issue6184.xaml:14]: Can you just use a page local to this test that gives instructions on how to make sure the test is passing?

**Review by PureWeen — CHANGES_REQUESTED** [review]
I think something with this fix is no longer consistent with the latest code

When I run the sample it crashes for me. I'm testing on an iPhone 12 with iOS 14.1

File: Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Issue6184.xaml (hunk 2 above)
Comment (pictos) [reply Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Issue6184.xaml:14]: Yes, I'll work on this.

**Comment by pictos** [comment]
@PureWeen I did some improvements in the code and for now, I removed support for iOS14. Looks like Apple changed the API and I'll need more time to find how to access the cells inside the More UITableView. I don't know if they closed the API or changed it to another place, or if is missing something in the bindings.

**Comment by PureWeen** [comment]
UI Tests
https://dev.azure.com/xamarin/public/_releaseProgress?_a=release-pipeline-progress&releaseId=2468

**Review by PureWeen — APPROVED** [review]

**Review by rmarinho — APPROVED** [review]

**Pull request merged**